source/js/components/petition/petition.jsx

```diff
@@ -348,6 +319,14 @@ export default class Petition extends React.Component {
     return input.match(/[^@]+@[^.@]+(\.[^.@]+)+$/) !== null;
   }
 
+  userElectedToDonate() {
+    this.setState({ showDonationModal: false });
+  }
+
+  userElectedToShare() {
+    this.setState({ showDonationModal: false });
```

Comment:
I assume we want analytics here, but I don't know which events and what payload those should have.